src/lockfiles.rs

```diff
@@ -0,0 +1,18 @@
+use anyhow::Context;
+use std::{collections::BTreeSet, fs::File, path::PathBuf};
+
+pub fn hash() -> anyhow::Result<blake3::Hash> {
+    let mut hasher = blake3::Hasher::new();
+
+    let lockfiles: BTreeSet<PathBuf> = crate::targets::targets()?
+        .into_iter()
+        .flat_map(|t| t.lockfiles())
```

Comment:
produces the same ordering each time?

Reply:
BTreeSet is an ordered set. BTreeMap is the ordered map version. https://doc.rust-lang.org/stable/std/collections/struct.BTreeSet.html